File: output/Person.m
#import "Person.h"

@implementation PersonBuilder
- (instancetype)init {
    if (self = [super init]) {
        _name = nil;
        _age = nil;
        _height = nil;
        _weight = nil;
        _alive = YES;
    }
    return self;
}
@end

@implementation Person

- (instancetype)initWithBuilder:(PersonBuilder *)builder {
    if (self = [super init]) {
        _name = builder.name;
        _age = builder.age;
        _height = builder.height;
        _weight = builder.weight;
        _alive = builder.alive;
    }
    return self;
}

- (PersonBuilder *)makeBuilder {
    PersonBuilder *builder = [PersonBuilder new];
    builder.name = _name;
    builder.age = _age;
    builder.height = _height;
    builder.weight = _weight;
    builder.alive = _alive;
    return builder;
}

- (instancetype)init {
    PersonBuilder *builder = [PersonBuilder new];
    return [self initWithBuilder:builder];
}

+ (instancetype)makeWithBuilder:(void (^)(PersonBuilder *))updateBlock {
    PersonBuilder *builder = [PersonBuilder new];
    updateBlock(builder);
    return [[Person alloc] initWithBuilder:builder];
}

- (instancetype)update:(void (^)(PersonBuilder *))updateBlock {
    PersonBuilder *builder = [self makeBuilder];
    updateBlock(builder);
    return [[Person alloc] initWithBuilder:builder];
}
@end

File: person.rb
headers <<-EOS
#import "SomeOtherFile.h
EOS
# headers에 string을 집어 넣는다. EOS -> end of string

#함수이름 인자, 인자(블록)
model "Person" do
  #함수이름  인자, 인자(해쉬) *해쉬 map & dictionary 같은 거
  property "name", type: "NSString *"
  property "age", type: "NSNumber *"
  property "height", type: "NSNumber *"
  property "weight", type: "NSNumber *"
  property "alive", type: "BOOL", default: "YES", getter: "isAlive"
end 
